addons/mail/static/src/discuss/core/common/discuss_core_common_service.js

```diff
@@ -109,11 +109,18 @@ export class DiscussCoreCommon {
                 thread: { id: channel_id, model: "discuss.channel" },
             });
         });
+        this.busService.subscribe("discuss.channel.member/read", (payload) => {
+            const member = this.store.ChannelMember.insert(payload.member);
+            if (member.eq(member.thread.selfMember)) {
+                member.thread.updateUnreadAndNeedactionCounters(member.new_message_separator);
+            }
```

Comment:
bus notif should send the new values directly instead

Reply:
I agree, on the other hand this is more complex than just sending the value on the bus: it's about removing every local increment of the unread counter or we could have race conditions such as:
- mark as unread, counter is sent on the bus
- new message is received, counter is incremented
- new counter is received from the first step, value is overwritten and incorrect

IMO fixing this counter is a task on its own. I think we do not want to send the new counter to each member during `message_post` so we have to keep a local compute on top of the server value?